projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-tool/tool.cpp

```diff
@@ -1567,7 +1575,102 @@ counter_record_callback(rocprofiler_dispatch_counting_service_data_t dispatch_da
         tool::write_ring_buffer(counter_record, domain_type::COUNTER_COLLECTION);
     }
 }
+std::optional<rocprofiler_spm_counter_config_id_t>
+get_spm_config(rocprofiler_agent_id_t agent_id)
+{
+    static const auto gpu_agents_counter_info = get_agent_counter_info();
+    static auto       agent_configs =
+        std::unordered_map<rocprofiler_agent_id_t, rocprofiler_spm_counter_config_id_t>{};
+
+    auto itr = agent_configs.find(agent_id);
+    if(itr != agent_configs.end()) return itr->second;
+
+    auto params = std::vector<rocprofiler_spm_parameter_t>{};
+    params.push_back(
+        {ROCPROFILER_SPM_PARAMETER_TYPE_TIMEOUT_MS, tool::get_config().spm_timeout_ms});
+    params.push_back(
+        {ROCPROFILER_SPM_PARAMETER_TYPE_BUFFER_SIZE, tool::get_config().spm_buffer_size * 1024});
+    params.push_back(
+        {ROCPROFILER_SPM_PARAMETER_TYPE_SCLK_COUNT, tool::get_config().spm_frequency_sclk});
```

Comment:
use a `auto params = std::array<rocprofiler_spm_parameter_t, 3>{  <param-1>, <param-2>, <param-3> };` here, you don't need a dynamic allocation and it is cleaner than all these push backs